src/ucs/arch/bitops.h

```diff
@@ -119,6 +119,86 @@ BEGIN_C_DECLS
 #define ucs_bitmap2idx(_map, _idx) \
     ucs_popcount((_map) & (UCS_MASK(_idx)))
 
+
+/**
+ * Count how many bits at the end of the buffer are equal to zero.
+ *
+ * @param [in] ptr         Pointer to the buffer.
+ * @param [in] bit_length  Total Buffer length (in bits).
+ *
+ * @return The number of trailing zero bits.
+ */
+static inline unsigned ucs_count_ptr_trailing_zero_bits(void *ptr,
+                                                        uint64_t bit_length)
+{
+    uint64_t idx = bit_length;
+    uint8_t  tmp = 0;
+
+    if (idx == 0) {
+        return 0;
+    }
+
+    /* Start from the end of the given buffer, with fractions of a bytes */
+    if ((idx % 8) != 0) {
+        tmp = *(uint8_t*)UCS_PTR_BYTE_OFFSET(ptr, idx / 8);
+        tmp &= ~UCS_MASK(8 - (idx % 8));
+        if (idx < 8) {
+            tmp |= UCS_BIT(idx % 8);
+        }
+        if ((idx < 8) || (tmp != 0)) {
+            return __builtin_ctz(tmp | (((uint32_t)-1) << 8));
+        }
+    }
```

Comment:
```
result = 0;
const uint64_t *p64 = UCS_PTR_BYTE_OFFSET(ptr, bit_length / 64)
while ((--ptr64 > ptr) && (*ptr64 == 0)) {
  result += 64;
}
return result + __builtin_clz(*ptr64)
```

Reply:
@yosefe  The suggested code only works if (bit_length % 8 == 0) - but we need to also support masks of 3 bits, for example.

Reply:
so let's use memrchr

Reply:
ok

Reply:
On second thought - I don't see how `memrchr()` can be applied here: where looking for the first **non-zero** byte, whereas `memrchr()` looks for the first byte matching a specific value...